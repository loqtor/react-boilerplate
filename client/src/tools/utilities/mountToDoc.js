/*
 * Utility to provide aXe with a fake DOM to allow for accessibility testing
 * on mounted components within our Jest tests
*/
import { mountWithIntl } from './enzymeIntlTestHelper';

let wrapper;

export default function mountToDoc(component) {
  const container = mountWithIntl(component);

  if (!wrapper) {
    wrapper = document.createElement('main');
    document.body.appendChild(wrapper);
  }

  wrapper.innerHTML = '';
  wrapper.appendChild(container.getDOMNode());

  return container;
}
