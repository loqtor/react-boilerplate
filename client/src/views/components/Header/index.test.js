import React from 'react';
import { StaticRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import axe from 'axe-core';

import mountToDoc from '../../../tools/utilities/mountToDoc';

import { Header } from './index';

const setup = (render, props) => {
  const toggleMobileMenuSpy = jest.fn();

  const defaultProps = {
    toggleMobileMenu: toggleMobileMenuSpy,
    mobileMenuIsOpen: false,
  };

  const component = render(
    <StaticRouter context={{}}>
      <Header {...defaultProps} {...props} />
    </StaticRouter>,
  );

  return {
    actual: component,
    toggleMobileMenuSpy,
  };
};

describe('index.test.jsx', () => {
  it('renders correctly', () => {
    const { actual } = setup(shallow, {});

    expect(toJson(actual)).toMatchSnapshot();
  });

  // it('renders with the correct class when the mobile menu is open', () => {
  //   const { actual } = setup(mount, { mobileMenuIsOpen: true });
  //
  //   expect(actual.find('.Hamburger').hasClass('is-open')).toBeTruthy();
  // });

  it('has no accessibility violations', (done) => {
    const { actual } = setup(mountToDoc, {});
    const componentNode = actual.getDOMNode();

    axe.run(componentNode, (err, { violations }) => {
      expect(err).toBe(null);
      expect(violations).toHaveLength(0);
      done();
    });
  });
});
