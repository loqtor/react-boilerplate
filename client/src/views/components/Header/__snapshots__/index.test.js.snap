// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.test.jsx renders correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Header
    mobileMenuIsOpen={false}
    toggleMobileMenu={[MockFunction]}
  />
</Router>
`;
